ra/ra.go

```diff
@@ -993,6 +1010,37 @@ func (ra *RegistrationAuthorityImpl) issueCertificate(
 			// This context is limited by the gRPC timeout.
 			_ = ra.publisher.SubmitToCT(context.Background(), cert.DER)
 		}()
+	} else if ra.ctpolicy != nil {
+		var ctCtx context.Context
+		currentDeadline, ok := ctx.Deadline()
+		if !ok {
+			// Current context doesn't have a deadline, this should
+			// never happen so it's a internal server error... but
+			// we already issued the cert so we can't fail out now.
+			// Just use a background context that lasts forever
+			// instead.
+			ctCtx = context.Background()
+		} else {
+			// NOTE: We want to check how putting the SCT submission/collection
+			// effects calls to IssueCertificate so we take the current context
+			// and allocate 80% of the remaining time to calling CTPolicy.GetSCTs.
+			// This way if we exceed the child context we won't time out the
+			// parent call and can still return the cert to the user.
+			until := time.Until(currentDeadline)
+			var cancel func()
+			ctCtx, cancel = context.WithTimeout(ctx, time.Duration(float64(until)*0.8))
+			defer cancel()
+		}
+		_, err := ra.ctpolicy.GetSCTs(ctCtx, cert.DER)
+		// The final cert has already been issued so actually return it to the
+		// user even if this fails since we aren't actually doing anything with
+		// the SCTs yet.
+		if err != nil {
+			ra.ctpolicyResults.With(prometheus.Labels{"result": "failure"}).Inc()
```

Comment:
Let's have a special error case for deadline exceeded. Also, instead of a counter, can we have a timing histogram?